src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmers.java

```diff
@@ -0,0 +1,135 @@
+        List<Kmer> killList = findBadGenomicKmers(ctx,getReference(),dict);
+        final File killFile = new File(outputFile);
+        try
+        {
+            final Writer writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(killFile)));
+
+            for ( Kmer kmer : killList )
+            {
+                writer.write(kmer.toString(KLEN));
+                writer.write('\n');
+            }
+            writer.close();
+        }
+        catch ( IOException e )
+        { throw new GATKException("Unable to write ubiquitous kmer kill file.", e); }
+    }
+
+    // Find high copy number kmers in the reference sequence
+    public static List<Kmer> findBadGenomicKmers( final JavaSparkContext ctx,
+                                                  final ReferenceMultiSource ref,
+                                                  final SAMSequenceDictionary readsDict )
+    {
+        // Write all the reference sequences to a text file as a series of 10,000-base lines.
+        // For each contig, records are written with a K-1 base overlap
+        // (i.e., the last 62 bases in line n match the first 62 bases in line n+1) so that we don't miss any kmers
+        // due to line breaks -- we can just kmerize each record.
+        File refSequenceFile;
+        try
+        {
+            refSequenceFile = File.createTempFile("genomic_sequence","txt");
+            refSequenceFile.deleteOnExit();
+            final OutputStream os = new BufferedOutputStream(new FileOutputStream(refSequenceFile));
+
+            final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(readsDict);
+            if ( dict == null )
+                throw new GATKException("No reference dictionary available.");
+
+            final int effectiveRecLen = REF_REC_LEN - KLEN + 1;
+            for ( final SAMSequenceRecord rec : dict.getSequences() )
+            {
+                final String seqName = rec.getSequenceName();
+                final int seqLen = rec.getSequenceLength();
+                final byte[] bases = ref.getReferenceBases(null,new SimpleInterval(seqName,1,seqLen)).getBases();
+                for ( int start = 0; start < seqLen; start += effectiveRecLen )
+                {
+                    os.write(bases,start,Math.min(REF_REC_LEN,seqLen-start));
+                    os.write('\n');
+                }
+            }
+
+            os.close();
+        }
+        catch ( IOException e )
+        { throw new GATKException("Unable to write reference text file for kmerizing.", e); }
+
+        // Turn that text file into an RDD, and do a classic map/reduce:
+        // Kmerize, mapping to a pair <kmer,1>, reduce by summing values by key, filter out <kmer,N> where N < 3,
+        // and collect the high frequency kmers back in the driver.
+        final int nParts = (int)(refSequenceFile.length()/CHUNK_SIZE + 1);
+        final JavaRDD<String> genomicBases = ctx.textFile(refSequenceFile.getPath(),nParts);
```

Comment:
Doesn't seem to be a performance killer.  Writing a 3Gb file in exactly the form it needs to be in so that it can be efficiently parallelized as a text file?  Takes less than a minute.  Text-file backing minimizes both driver and executor memory requirements.  Also, note that I use K-1 overlaps to simplify kmerization.  Seems like a pretty good way of RDDing the reference sequence to me, though I remain open to other ideas.